src/migrations/20240702000000-remove_national_center_ars.js

```diff
@@ -0,0 +1,553 @@
+        ;
+        -- remove ARGs -------------------
+        DROP TABLE IF EXISTS deleted_args;
+        CREATE TEMP TABLE deleted_args AS
+        WITH deletes AS (
+          DELETE FROM "ActivityReportGoals"
+          USING args_to_delete
+          WHERE id = argid
+          RETURNING
+          id,
+          "goalId" gid
+        )
+        SELECT id, gid FROM deletes
+        ;
+
+        -- Create the orphaned Goal deletions list
+        DROP TABLE IF EXISTS goals_to_delete;
+        CREATE TEMP TABLE goals_to_delete
+        AS
+        SELECT DISTINCT gid
+        FROM deleted_args dargs
+        JOIN "Goals" g
+          ON gid = g.id
+        WHERE (g."isRttapa" IS NULL OR g."isRttapa" != 'Yes')
+          AND g."isFromSmartsheetTtaPlan" != TRUE
+          AND g."createdVia" != 'merge'
+        EXCEPT 
+        SELECT gid
+        FROM (
+          SELECT DISTINCT "goalId" gid
+          FROM "ActivityReportGoals"
+          UNION
+          SELECT DISTINCT "goalId"
+          FROM "Objectives"
+          UNION
+          SELECT DISTINCT "goalId"
+          FROM "EventReportPilotGoals"
+        ) keepers
+        ;
+        -- Remove Goal link records: -------------
+        DROP TABLE IF EXISTS deleted_goalcollaborators;
+        CREATE TEMP TABLE deleted_goalcollaborators AS
+        WITH deletes AS (
+          DELETE FROM "GoalCollaborators"
+          USING goals_to_delete
+          WHERE "goalId" = gid
+          RETURNING
+          id
+        )
+        SELECT id FROM deletes
+        ;
+        DROP TABLE IF EXISTS deleted_goalfieldresponses;
+        CREATE TEMP TABLE deleted_goalfieldresponses AS
+        WITH deletes AS (
+          DELETE FROM "GoalFieldResponses"
+          USING goals_to_delete
+          WHERE "goalId" = gid
+          RETURNING
+          id
+        )
+        SELECT id FROM deletes
+        ;
+        DROP TABLE IF EXISTS deleted_goalresources;
+        CREATE TEMP TABLE deleted_goalresources AS
+        WITH deletes AS (
+          DELETE FROM "GoalResources"
+          USING goals_to_delete
+          WHERE "goalId" = gid
+          RETURNING
+          id,
+          "resourceId" resourceid
+        )
+        SELECT id, resourceid FROM deletes
+        ;
+        DROP TABLE IF EXISTS deleted_goalstatuschanges;
+        CREATE TEMP TABLE deleted_goalstatuschanges AS
+        WITH deletes AS (
+          DELETE FROM "GoalStatusChanges"
+          USING goals_to_delete
+          WHERE "goalId" = gid
+          RETURNING
+          id,
+          "goalId" gid
+        )
+        SELECT id, gid FROM deletes
+        ;
+        -- remove Goals -------------------
+        DROP TABLE IF EXISTS deleted_goals;
+        CREATE TEMP TABLE deleted_goals AS
+        WITH deletes AS (
+          DELETE FROM "Goals"
+          USING goals_to_delete
+          WHERE id = gid
+          RETURNING
+          id,
+          "goalTemplateId" gtid
+        )
+        SELECT id, gtid FROM deletes
```

Comment:
OnApprovedAR and OnAR need to be updated for any goals not deleted

Reply:
`onApprovedAR` and `onAR` are now fixed at the end